Update SB's CI distro coverage guidelines (#4516)
dotnet/source-build: Documentation/ci-platform-coverage-guidelines.md

1. Only test [distros that are officially supported by .NET](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md#net-supported-os-policy).
Community supported distros will not be covered.

## Distro Versions

> 1. Prefer testing latest LTS version if LTS is supported else the latest version.

Reviewer: ```suggestion
1. Prefer testing latest LTS version. If LTS is unsupported, use the latest version instead.
```

> 1. Drop distro versions that will be or nearing EOL on the .NET release day.

Reviewer: ```suggestion
1. Drop distro versions that will be or are nearing EOL on the .NET release day.
```

> 1. Stop updating the test matrix during the last 6 months of support for a .NET version as new distro

Reviewer: ```suggestion
1. Stop updating the test matrix in the last 6 months of support for a .NET version as new distro
```
